core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -164,6 +164,18 @@ private[v1] class ApiRootResource extends UIRootFromServletContext {
     }
   }
 
+  @Path("applications/{appId}/download")
+  def getEventLogs(
+    @PathParam("appId") appId: String): EventLogDownloadResource = {
+    new EventLogDownloadResource(uiRoot, appId, None)
+  }
+
+  @Path("applications/{appId}/{attemptId}/download")
```

Comment:
Ah, that's too bad; wish I had caught that earlier, it's probably too late now. :-/

Anyway, I have a slight preference for `/logs` as the URI, since REST generally discourages using verbs.